cpp/src/arrow/acero/hash_aggregate_test.cc

```diff
@@ -759,11 +741,22 @@ TEST(RowSegmenter, MultipleSegments) {
                   {4, 2, false, false},
                   {6, 2, false, false},
                   {8, 1, true, false}});
+  };
+  for (int num_keys = 1; num_keys <= 2; ++num_keys) {
+    test_with_keys(num_keys, ArrayFromJSON(int32(), "[1, 1, 2, 5, 3, 3, 5, 5, 4]"));
+    test_with_keys(
+        num_keys,
+        ArrayFromJSON(fixed_size_binary(2),
+                      R"(["aa", "aa", "bb", "ee", "cc", "cc", "ee", "ee", "dd"])"));
+    test_with_keys(num_keys, DictArrayFromJSON(dictionary(int8(), utf8()),
+                                               "[0, 0, 1, 4, 2, 2, 4, 4, 3]",
+                                               R"(["a", "b", "c", "d", "e"])"));
```

Comment:
I added the "non-constant batch" cases here instead of `TestRowSegmenterConstantBatch`. Because `TestRowSegmenterConstantBatch` tests scalars which are by definition cannot be "non-constant".

Reply:
FYI @pitrou

Reply:
Great, thank you!